crates/transaction-pool/src/pool/parked.rs

```diff
@@ -298,16 +298,73 @@ impl<T: PoolTransaction> ParkedPool<BasefeeOrd<T>> {
     /// Removes all transactions and their dependent transaction from the subpool that no longer
     /// satisfy the given basefee.
     ///
+    /// Legacy method maintained for compatibility with read-only queries.
+    /// For basefee enforcement, prefer `enforce_basefee_with_handler` for better performance.
+    ///
     /// Note: the transactions are not returned in a particular order.
+    #[cfg(test)]
     pub(crate) fn enforce_basefee(&mut self, basefee: u64) -> Vec<Arc<ValidPoolTransaction<T>>> {
-        let to_remove = self.satisfy_base_fee_ids(basefee as u128);
+        let mut removed = Vec::new();
+        self.enforce_basefee_with_handler(basefee, |tx| {
+            removed.push(tx);
+        });
+        removed
+    }
+
+    /// Removes all transactions from this subpool that can now afford the given basefee,
+    /// invoking the provided handler for each transaction as it is removed.
+    ///
+    /// This method enforces the basefee constraint by identifying transactions that now
+    /// satisfy the basefee requirement (typically after a basefee decrease) and processing
+    /// them via the provided transaction handler closure.
+    ///
+    /// Respects per-sender nonce ordering: if the lowest-nonce transaction for a sender
+    /// still cannot afford the basefee, higher-nonce transactions from that sender are skipped.
+    ///
+    /// Note: the transactions are not returned in a particular order.
+    pub(crate) fn enforce_basefee_with_handler<F>(&mut self, basefee: u64, mut tx_handler: F)
```

Comment:
can drop _handler here imo
```suggestion
    pub(crate) fn enforce_basefee_with<F>(&mut self, basefee: u64, mut tx_handler: F)
```

Reply:
that looks alot better, thanks!